### PR TITLE
Update to reflect current best practice of not splitting out M/V/U/F mo…

### PR DESCRIPTION
**Pull request by avh4**
…dules for every page

### CODE REVIEW

**Comment by avh4** [comment]
Tried to make this from https://trello.com/c/Qpxu4HRd/32-pr-to-elm-style-guide-about-consolidating-model-update-view

**Review by mthadley — CHANGES_REQUESTED** [review]
I think we should also update the section that describes `Html.programWithFlags`. It still refers to the old module structure:

> It calls `Html.programWithFlags` with:
> - `init`, defined in `Main`, runs `Flags.decodeFlags` and turns the resulting `Flags` type into a > `Model`.
> - `update` is `Update.update >> batchUpdate`. See [NoRedInk/rocket-update](https://github.com/NoRedInk/rocket-update) for details on `batchUpdate`.
> - `view` is simply `View.view`.
> - `subscriptions`, defined in `Main`, contains any subscriptions this app relies on.

**Review by mthadley — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -117,55 +113,44 @@ Don't do: `Nri/Leaderboard/Main.elm` - the filename `Main.elm` is reserved for e
...
 - Main.elm
-- Model.elm
-- Update.elm
-- View.elm
-- Flags.elm
+    - `type alias Flags = { ...a record that directly corresponds to the JSON page data... }`
+    - `flagsDecoder : Json.Decode.Decoder Flags`
+    - `type alias Model = { ...a record with fields... }`
+    - `init : Flags -> (Model, Cmd Msg)`
+    - `type Msg = ... variants for each possible message ...
```
Comment (dphm) [inline README.md]: Missing a backtick

**Comment by avh4** [comment]
(waiting for @tesk9 to have a chance to look before merging)

File: README.md [hunk 2]
```diff
@@ -117,55 +113,44 @@ Don't do: `Nri/Leaderboard/Main.elm` - the filename `Main.elm` is reserved for e
...
 
-**`Update.elm`** contains our update code. This includes the `Msg` types for our view. Inside here most of our business logic lives.
+**`Msg`, `update`** contains our update code. Inside here most of our business logic lives.
 
-Inside **`View.elm`**, we define the view for our model and set up any event handlers we need.
+Inside **`view`**, we define the view for our model and set up any event handlers we need.
 
-**`Flags.elm`** contains a decoder for the flags of the app. We aim to keep our decoders basic and so decode into a special `Flags` type that mirrors the structure of the raw JSON instead of the structure of the `Model` type. The `Flags` and `Model` modules should not depend on each other.
+**`Flags`, `flagsDecoder`** is a decoder for the flags of the app. We aim to keep our decoders basic and so decode into a special `Flags` type that mirrors the structure of the raw JSON instead of the structure of the `Model` type. The `Flags` and `Model` modules should not depend on each other.
```
Comment (tesk9) [inline README.md]: I think calling this `flagsDecoder` is a change. In the monolith, I searched for `decoder :` and `flagsDecoder :`. There are only 14 results for `flagsDecoder` (and too of those are from `Nri.Program`).

I wonder if we should talk about changing to `flagsDecoder` in a separate convo from the modules convo. But also, not a big deal.

**Review by tesk9 — APPROVED** [review]
Looks great! One comment about a name change. Thanks for doing this!

File: README.md (hunk 2 above)
Comment (avh4) [reply README.md]: I'm fine with `decoder` here if that's what's in use -- I didn't actually check; I just guessed that it was being called `flagsDecoder` :)

Comment (avh4) [reply README.md]: updated

**Pull request merged**